Source/WTF/wtf/Int128.h

```diff
@@ -1264,10 +1271,72 @@ using UInt128 = UInt128Impl;
 using Int128 = Int128Impl;
 #endif
 
+template<> struct DefaultHash<UInt128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const UInt128& a, const UInt128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+template<> struct DefaultHash<Int128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const Int128& a, const Int128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+#if !HAVE(INT128_T)
+template<> struct HashTraits<UInt128> : public SimpleClassHashTraits<UInt128> { };
+template<> struct HashTraits<Int128> : public SimpleClassHashTraits<Int128> { };
+#endif
+
 WTF_EXPORT_PRIVATE void printInternal(PrintStream&, UInt128);
 WTF_EXPORT_PRIVATE void printInternal(PrintStream&, Int128);
 
 }  // namespace WTF
 
+#if !HAVE(INT128_T)
+namespace IPC {
+template<> struct ArgumentCoder<WTF::UInt128> {
+    template<typename Encoder> static void encode(Encoder& encoder, const WTF::UInt128& i)
+    {
+        encoder << WTF::UInt128High64(i);
+        encoder << WTF::UInt128Low64(i);
+    }
+    template<typename Decoder> static std::optional<WTF::UInt128> decode(Decoder& decoder)
+    {
+        std::optional<uint64_t> high;
+        decoder >> high;
+        if (!high)
+            return std::nullopt;
+
+        std::optional<uint64_t> low;
+        decoder >> low;
+        if (!low)
+            return std::nullopt;
+
+        return WTF::MakeUInt128(*high, *low);
```

Comment:
Ditto. This function should not be used. Do usual integer things.
```
(static_cast<UInt128>(*high) << 64) | *low
```

Reply:
This is for a platform with `!HAVE(INT128_T)`, so `static_cast<UInt128>` won't work right?